src/Console/Mappings/IndexCopyCommand.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace DesignMyNight\Elasticsearch\Console\Mappings;
+
+use Illuminate\Support\Facades\DB;
+
+/**
+ * Class IndexCopyCommand
+ *
+ * @package DesignMyNight\Elasticsearch\Console\Mappings
+ */
+class IndexCopyCommand extends Command
+{
+    /**
+     * @var string $description
+     */
+    protected $description = 'Populate an index with all documents from another index';
+
+    /**
+     * @var string $signature
+     */
+    protected $signature = 'index:copy {from} {to}';
+
+    /**
+     * @var int
+     */
+    protected $batchSize = 1000;
+
+    /**
+     * Execute the console command.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        ['from' => $fromIndex, 'to' => $toIndex] = $this->arguments();
+
+        $bulkParams = ['body' => []];
+        $connection = DB::connection('elasticsearch');
+
+        $query = [
+            'index' => $fromIndex,
+            'body' => [
+                'query' => [
+                    'match_all' => (object) []
+                ],
+            ],
+        ];
+
+        $numResultsQuery = $query;
+        $numResultsQuery['size'] = 0;
+
+        $numResults = $connection->select($numResultsQuery)['hits']['total'];
+        $bar = $this->output->createProgressBar($numResults);
+        $bar->start();
+
+        foreach ($connection->cursor($query) as $i => $doc) {
+            $indexParams = [
+                'index' => [
+                    '_index' => $toIndex,
+                    '_type' => $doc['_type'],
+                    '_id' => $doc['_id'],
+                ],
+            ];
+
+            if (isset($doc['_parent'])) {
+                $indexParams['index']['parent'] = $doc['_parent'];
+            }
+
+            $bulkParams['body'][] = $indexParams;
+            $bulkParams['body'][] = $doc['_source'];
+
+            if (($i + 1) % $this->batchSize === 0) {
+                $responses = $connection->insert($bulkParams);
+                $bulkParams = ['body' => []];
+
+                $bar->advance($this->batchSize);
```

Comment:
Depending on how OCD you are, say there is only 1500 documents to index, once it reaches 1500 it will advance the progress bar by another 1000, so it will end up saying 2000/1500...just so you know...

Reply:
Or does the `$bar->finish();` set the bar to the correct number?

Reply:
I'm definitely OCD enough to want to fix that. It should only ever do `$bar->advance()` inside the `if` that checks that the current index is evenly divisible by the batch size, then once it's out of the loop if there's any docs in the `$bulkParams` array it will index those (so the last 500 in your example) then do `$bar->finish()` which jumps it to 100%. Does that look right to you?

Reply:
LGTM!